torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2908,6 +2923,27 @@ def create_symintnode(
             if hint is not None:
                 assert int(sym) == hint
             out = int(sym)
+        elif is_nested_int(hint):
+            # See Note [Recursive fakification]
+            fake_vec = metafy_fn(
+                nested_int.tensor,
+                torch._dynamo.source.SymNodePropertySource(source, "get_tensor"),
+            )
+            coeff = hint.node.nested_int_coeff()
+            nested_int = SymInt(
+                SymNode(
+                    sym, self, int,
+                    # The hint is a non-symbolic nested int with FakeTensor as its tensor.
+                    # if for_hint is specified, we go through normal path even though
+                    # we passed in a FakeTensor - can we do this cleaner?
+                    torch.nested._internal.nested_tensor.get_nested_symint(fake_vec, coeff=coeff, for_hint=True),
```

Comment:
what does `for_hint` mean?

Reply:
Probably not very aptly named lol, but in this stack, symbolic nested int holds a non-symbolic nested int whose hint is actually a FakeTensor, and this is a way to reuse the existing code path to do that.

Ordinarily, if I'm getting the nested int for a particular offsets that is Fake, I want to instead call the get_nested_symint method on the FakeTensor, because FakeTensor is responsible for caching the symbolic nested int directly on itself.

We pass this in extra arg to distinguish the special case where I am creating the symbolic nested int for the first time, and actually want to create a NestedIntNode.